src/vs/editor/contrib/snippet/common/snippet.ts

```diff
@@ -55,7 +55,39 @@ export class CodeSnippet implements ICodeSnippet {
 	}
 
 	static fromEmmet(template: string): CodeSnippet {
-		return EmmetSnippetParser.parse(template);
+		//return EmmetSnippetParser.parse(template);
+
+		let matchFinalStops = template.match(/\$\{0\}/g);
+		if (!matchFinalStops || matchFinalStops.length === 1) {
+			return CodeSnippet.fromTextmate(template);
+		}
+
+		// Emmet sometimes returns snippets with multiple ${0}
+		// In such cases, replace ${0} with incremental tab stops
+
+		let maxTabStop = -1;
+		let matches = [];
+		let regex = /\$\{(\d+)\}/g;
+		while (matches = regex.exec(template)) {
+			let tabStop = Number(matches[1]);
+			if (tabStop > maxTabStop) {
+				maxTabStop = tabStop;
+			}
+		}
+		regex = /\$\{(\d+):[^\}]+\}/g;
+		matches = [];
+		while (matches = regex.exec(template)) {
+			let tabStop = Number(matches[1]);
+			if (tabStop > maxTabStop) {
+				maxTabStop = tabStop;
+			}
+		}
```

Comment:
If this is becoming too complex, e.g `$1`, with escaped `}`, or nested placeholder (stretch, like `foo ${1:\}}` or `foo${2:b${1:ar}$0`), we should consider using the (new) snippet parser (`SnippetParser`) which allows you read all placeholders. Tho I'd say that depends on the Emmet-TextMate-subset